### ISSUE TITLE
`have_enqueued_job` clashes with `rspec-rails` `3.4.0`

### ISSUE THREAD

**Issue by JonRowe** [philostler/rspec-sidekiq#83]
In `rspec-rails` we implemented support for asserting on wether or not an `ActiveJob` has been queued, unfortunately we used `have_enqueued_job` as the name for this matcher as we were unaware of this gem. This means that `rspec-sidekiq` will now clash with `rspec-rails`.

Given that our matcher is more generic I propose that the `have_enqueued_job` matcher in `rspec-sidekiq` be renamed to `have_enqueued_sidekiq_job` and then aliased back to `have_enqueued_job` when it's not otherwise defined. (We use the alias trick a fair bit in `rspec-expectations` to provide extra grammar for matchers so it's a pretty accepted technique).

I'd be happy to work on a PR for this if it's acceptable to you :)

**Comment by mhenrixon** [comment]
:+1:

**Comment by aprescott** [comment]
I see @mrbrdo issued #84 for this fix. Until that gets reviewed, merged and included in a release, is there a temporary work-around I can use to unblock gem updates?

**Comment by aprescott** [comment]
Think I got one, actually:

```ruby
# Temporarily work around a clash between rspec-rails 3.4.0 and
# rspec-sidekiq because of new ActiveJob matchers.
#
# See philostler/rspec-sidekiq#83
module RSpec
  module Sidekiq
    module Matchers
      def have_enqueued_sidekiq_job(*expected_arguments)
        HaveEnqueuedSidekiqJob.new expected_arguments
      end

      HaveEnqueuedSidekiqJob = HaveEnqueuedJob
    end
  end
end
```

**Comment by mrbrdo** [comment]
Or just use my branch in your gemfile.

**Comment by aprescott** [comment]
Git refs in a Gemfile lead to bulky `vendor/cache` directories, so I try to avoid them.

**Comment by fabn** [comment]
@philostler any ETA for this? I'd like to upgrade RSpec but this issue prevent me to do that.

**Comment by philostler** [comment]
Been busy with work etc recently, will review all PRs and Issues very soon.

**Comment by bkoltai** [comment]
+1 for PR!

**Comment by nickcoyne** [comment]
Was this not merged?

**Comment by philostler** [comment]
This should be fixed with #95 now
